### PR TITLE
Override Jira teams label via config

### PR DESCRIPTION
**Pull request by fairclothjm**
Test run: https://github.com/hashicorp/vault-plugin-vaulteco-testing/actions/runs/9880157535

### CODE REVIEW

**Review by tomcf-hcp — APPROVED** [review]
Looks good to me

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -53,6 +53,24 @@ jobs:
...
 
+        curl -o repo-teams.txt https://raw.githubusercontent.com/hashicorp/vault-workflows-common/main/repo-teams.txt
+
+        echo "::debug::Determining team labels"
+        org_with_repo=${{ github.event.repository.name }}
+        repo="${org_with_repo#hashicorp/}"
+        label="$(grep --max-count 1 $repo repo-teams.txt | awk '{print $2}')"
+        # append team label to array
+        teams_field='[]'
+        teams_field="$(echo "$teams_field" | jq --compact-output --arg tt "$label" '. |= . + [$tt]')"
```
Comment (fairclothjm) [inline .github/workflows/jira.yaml:64]: This jq nonsense is probably not necessary, I just copied it from our previous override logic that used a for loop. Happy to change it if anyone feels strongly.

File: repo-teams.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+vault-plugin-database-redis              vault-eco
+vault-plugin-database-redis-elasticache  vault-eco
+vault-plugin-database-snowflake          vault-eco
+vault-plugin-secrets-ad                  vault-eco
+vault-plugin-secrets-alicloud            vault-eco
+vault-plugin-secrets-azure               vault-eco
+vault-plugin-secrets-gcp                 vault-eco
+vault-plugin-secrets-gcpkms              vault-eco
+vault-plugin-secrets-kubernetes          vault-eco
+vault-plugin-secrets-kv                  core
```
Comment (tvoran) [inline repo-teams.txt:23]: This team may need to be `foundations`?

Comment (fairclothjm) [reply repo-teams.txt:23]: oh, yeah I forgot to check on that. Thanks!

File: .github/workflows/jira.yaml (hunk 1 above)
Comment (fairclothjm) [reply .github/workflows/jira.yaml]: This bugged me so I changed it.

File: repo-teams.txt (hunk 2 above)
Comment (fairclothjm) [reply repo-teams.txt:23]: I asked in the vault slack. I will merge this now and we can update when I get a response.

**Pull request merged**